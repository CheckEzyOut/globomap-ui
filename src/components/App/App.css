/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

body {
  border: 0;
  padding: 0;
  margin: 0;
  background-color: white;
  font-family: "opensans-light", Helvetica, sans-serif;
}

a {
  color: #288edf
}

button {
  cursor: pointer;
  outline: none;
}

button:hover {
  opacity: 0.90;
}

button:disabled {
  cursor: default;
  opacity: 0.3;
}

.has-id {
  position: absolute;
  left: 0;
  top: 0;
  height: 10px;
  width: 10px;
  user-select: none;
  z-index: 10;
}
