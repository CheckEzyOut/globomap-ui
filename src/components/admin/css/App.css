/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.admin-content,
.admin-content-top,
.admin-content-middle,
.admin-content-base {
  box-sizing: border-box;
}

.admin-content {
  display: flex;
  flex-direction: column;
  overflow: hidden;
  justify-content: space-between;

  position: absolute;
  top: 60px;
  right: 10px;
  left: 10px;
  bottom: 10px;
  border: 1px solid #9daca9;
  border-radius: 4px;
  box-shadow: 0 3px 25px rgba(0,0,0,0.2);
}

.admin-content-top {
  background-color: #d9dddd;
  height: 32px;
  left: 11px;
  padding: 4px;
  position: fixed;
  right: 11px;
}

.admin-content-middle {
  bottom: 33px;
  overflow-y: scroll;
  padding: 20px;
  position: absolute;
  top: 33px;
  width: 100%;
}

.admin-content-base {
  background-color: #9daca9;
  bottom: 11px;
  height: 32px;
  left: 11px;
  padding: 4px;
  position: fixed;
  right: 11px;
}


.bs-ui-form-control {
  font-size: 0.875rem;
  margin-bottom: 20px;
  position: relative;
}

.bs-ui-form-control__label {
  color: #666;
  display: block;
  margin-bottom: 10px;
}

.bs-ui-form-control__field {
  background-color: #fff;
  border: solid 2px #ccc;
  border-radius: 4px;
  box-sizing: border-box;
  color: #333;
  height: 40px;
  outline: 0;
  padding: 7px 8px 8px;
  transition: border-color .15s cubic-bezier(0,0,.2,1);
  width: 100%
}
