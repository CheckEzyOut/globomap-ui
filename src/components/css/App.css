/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

body {
  border: 0;
  padding: 0;
  margin: 0;
  background-color: #f8f8f8;
  font-family: sans-serif;
}

a {
  color: #288edf
}

.graph-color0,
.graph-color0:hover { background-color: #347eb9 }

.graph-color1,
.graph-color1:hover { background-color: #ccaf6f }

.graph-color2,
.graph-color2:hover { background-color: #d43e5e }

.graph-color3,
.graph-color3:hover { background-color: #2b9e63 }
