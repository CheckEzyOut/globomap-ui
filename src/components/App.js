/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

/* global _ */

import React, { Component } from 'react';
import io from 'socket.io-client';
import Header from './Header';
import Tools from './Tools';
import SearchContent from './SearchContent';
import Stage from './Stage';
import Info from './Info';
import PopMenu from './PopMenu';
import { traverseItems, uuid, sortByName } from '../utils';
import './css/App.css';

function uiSocket() {
  var uiSocket = io();
  uiSocket.on('error', function(err){
    window.location.reload();
  });
  return uiSocket;
}

class App extends Component {

  constructor(props) {
    super(props);
    this.socket = uiSocket();

    this.state = {
      currentNode: false,
      graphs: [],
      enabledCollections: [],
      collectionsByGraphs: {},
      collections: [],
      nodes: [],
      stageNodes: [],
      hasId: false
    };

    this.findNodes = this.findNodes.bind(this);
    this.getNode = this.getNode.bind(this);
    this.addNodeToStage = this.addNodeToStage.bind(this);
    this.stageHasNode = this.stageHasNode.bind(this);
    this.getGraphsAndCollections = this.getGraphsAndCollections.bind(this);
    this.getCollectionByGraphs = this.getCollectionByGraphs.bind(this);
    this.setStageNodes = this.setStageNodes.bind(this);
    this.setCurrent = this.setCurrent.bind(this);
    this.clearCurrent = this.clearCurrent.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.clearStage = this.clearStage.bind(this);
    this.onToggleGraph = this.onToggleGraph.bind(this);
    this.removeNode = this.removeNode.bind(this);
    this.handleDoubleClick = this.handleDoubleClick.bind(this);
    this.clearInfo = this.clearInfo.bind(this);
  }

  render() {
    return (
      <div className="main">
        <span className="main-xxxx"
              onDoubleClick={this.handleDoubleClick}>&nbsp;</span>
        <Header ref={(header) => {this.header = header}}
                graphs={this.state.graphs}
                enabledCollections={this.state.enabledCollections}
                clearStage={this.clearStage}
                clearCurrent={this.clearCurrent}
                collections={this.state.collections}
                findNodes={this.findNodes}
                onToggleGraph={this.onToggleGraph}
                clearInfo={this.clearInfo}
                searchContent={this.searchContent} />

        <Tools currentNode={this.state.currentNode}
               stageNodes={this.state.stageNodes}
               setStageNodes={this.setStageNodes}
               popMenu={this.popMenu} />

        <SearchContent ref={(searchContent) => {this.searchContent = searchContent}}
                       nodes={this.state.nodes}
                       findNodes={this.findNodes}
                       addNodeToStage={this.addNodeToStage}
                       currentNode={this.state.currentNode}
                       enabledCollections={this.state.enabledCollections}
                       header={this.header} />

        <Stage graphs={this.state.graphs}
               stageNodes={this.state.stageNodes}
               currentNode={this.state.currentNode}
               clearCurrent={this.clearCurrent}
               removeNode={this.removeNode}
               setCurrent={this.setCurrent}
               hasId={this.state.hasId} />

        <Info ref={(Info) => {this.info = Info}}
              getNode={this.getNode}
              stageNodes={this.state.stageNodes}
              graphs={this.state.graphs}
              collectionsByGraphs={this.state.collectionsByGraphs}
              stageHasNode={this.stageHasNode}
              addNodeToStage={this.addNodeToStage}
              clearCurrent={this.clearCurrent}
              currentNode={this.state.currentNode}
              hasId={this.state.hasId} />

        <PopMenu ref={(popMenu) => {this.popMenu = popMenu}}
                 currentNode={this.state.currentNode} />
      </div>
    );
  }

  getEdgeLinks(graph) {
    let collections = [];

    graph.links.forEach((edge) => {
      edge.from_collections.forEach((key) => {
        collections.push(key);
      })
      edge.to_collections.forEach((key) => {
        collections.push(key);
      })
    });

    return collections;
  }

  getGraphsAndCollections() {
    this.socket.emit('getcollections', {}, (data) => {
      this.setState({ collections: data });
    });

    this.socket.emit('getgraphs', {}, (items) => {
      let graphs = [];
      let enabledCollections = [];
      let collectionsByGraphs = {};
      let collections = [];
      items = sortByName(items);

      items.forEach((item, index) => {
        collections = this.getEdgeLinks(item);

        graphs.push({
          name: item.name,
          colorClass: 'graph-color' + index,
          enabled: true
        });

        enabledCollections = enabledCollections.concat(collections);
        collectionsByGraphs[item.name] = _.uniq(collections);
      });

      this.setState({
        graphs: graphs,
        enabledCollections: _.uniq(enabledCollections),
        collectionsByGraphs: collectionsByGraphs
      });
    });
  }

  getCollectionByGraphs(graphsCopy, fn) {
    let enabledCollections = [];

    this.socket.emit('getgraphs', {}, (items) => {
      items = sortByName(items);

      items.forEach((item, index) => {
        if (!graphsCopy[index].enabled) {
          return;
        }
        enabledCollections = enabledCollections.concat(
          this.getEdgeLinks(item));
      });

      fn(_.uniq(enabledCollections));
    });
  }

  addNodeToStage(node, parentUuid, makeCurrent=false) {
    if(this.stageHasNode(node._id, parentUuid)) {
      let n = this.getNode(node, parentUuid);
      this.setCurrent(n);
      return;
    }

    node.uuid = uuid();
    node.items = node.items || [];
    let currentNodes = this.state.stageNodes.slice();

    if(parentUuid) {
      traverseItems(currentNodes, (n) => {
        if(n.uuid === parentUuid) {
          n.items.push(node);
        }
      });
    } else {
      node.root = true;
      currentNodes = [node];
    }

    return this.setState({ stageNodes: currentNodes }, () => {
      if(makeCurrent) {
        this.setCurrent(node);
      }
    });
  }

  stageHasNode(nodeId, parentUuid) {
    let stageNodes = this.state.stageNodes,
        ids = stageNodes.map(n => n._id);

    if(parentUuid !== undefined) {
      traverseItems(stageNodes, (n) => {
        if(n.uuid === parentUuid) {
          ids = n.items.map(n => n._id);
        }
      });
    }

    return !(ids.indexOf(nodeId) < 0);
  }

  getNode(node, parentUuid) {
    let stageNodes = this.state.stageNodes.slice(),
        nodeFound = false;

    traverseItems(stageNodes, (n) => {
      if(n.uuid === node.uuid) {
        nodeFound = n;
      }
    });

    if(!nodeFound && parentUuid !== undefined) {
      traverseItems(stageNodes, (n) => {
        if(n.uuid === parentUuid) {
          for(let i in n.items) {
            if(node._id === n.items[i]._id) {
              nodeFound = n.items[i];
            }
          }
        }
      });
    }

    return nodeFound;
  }

  removeNode(node) {
    let stageNodes = this.state.stageNodes.slice();
    let i = stageNodes.findIndex((n) => {
        return n.uuid === node.uuid;
    });

    if (i >= 0) {
      stageNodes.splice(i, 1);
    } else {
      traverseItems(stageNodes, (n) => {
        let j = n.items.findIndex((n) => {
          return n.uuid === node.uuid;
        });
        if (j >= 0) {
          n.items.splice(j, 1);
        }
      });
    }

    this.setState({ stageNodes: stageNodes }, () => {
      this.clearCurrent();
    });
  }

  clearStage() {
    return this.setState({ stageNodes: [] }, () => {
      this.clearCurrent();
    });
  }

  clearInfo(fn) {
    this.info.resetByGraph(fn);
  }

  findNodes(query, co, per_page, page, fn) {
    if (co !== undefined && !co instanceof Array)  {
      console.log('The 2nd argument must be an Array');
      return;
    }

    this.socket.emit('findnodes', { query: query, collections: co, per_page: per_page, page: page }, (data) => {
      if (fn === undefined) {
        fn = () => {};
      }

      this.setState({ nodes: data.documents}, fn(data));
    });
  }

  setStageNodes(stageNodes, fn) {
    this.setState({ stageNodes }, fn);
  }

  setCurrent(node, fn) {
    this.setState({ currentNode: { _id: node._id, uuid: node.uuid } }, fn);
  }

  clearCurrent(fn) {
    this.setState({ currentNode: false }, fn);
  }

  handleKeyDown(event) {
    if (event.key === 'Escape') {
      event.preventDefault();
      this.clearCurrent();
    }
  };

  onToggleGraph(event, graphName) {
    event.stopPropagation();
    let graphsCopy = this.state.graphs.map((graph) => {
      if(graph.name === graphName) {
        graph.enabled = !graph.enabled;
      }
      return graph;
    });

    this.getCollectionByGraphs(graphsCopy, (enabledCollections) => {
      this.setState({
        graphs: graphsCopy,
        enabledCollections: enabledCollections
      }, () => {
        this.info.onTraversalSearch();
      });
    });
  }

  handleDoubleClick() {
    this.setState({
      hasId: !this.state.hasId
    });
  }

  componentDidMount() {
    this.getGraphsAndCollections();
    document.addEventListener('keydown', _.throttle(this.handleKeyDown, 100))
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleKeyDown)
    this.socket.disconnect();
  }

}

export { App as default, uiSocket };
