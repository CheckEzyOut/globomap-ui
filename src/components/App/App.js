/*
Copyright 2019 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import _ from "lodash";
import 'tippy.js/dist/tippy.css';
import React from 'react';
import { connect } from 'react-redux';
import {
  fetchGraphs,
  fetchCollections,
  fetchEdges,
  fetchQueries,
  getServerData,
  toggleHasId } from '../../redux/modules/app';
import {
  clearCurrentNode,
  resetSubNodes } from '../../redux/modules/nodes';
import {
  Favorites,
  Modal,
  Reports,
  Sidebar,
  Search,
  Stage,
  Loading,
  TabContent } from '../';
import './App.css';

export class App extends React.Component {

  constructor(props) {
    super(props);
    this.handleKeyDown = this.handleKeyDown.bind(this);
  }

  handleKeyDown(event) {
    if (event.key === 'Escape') {
      event.preventDefault();
      this.props.clearCurrentNode();
      this.props.resetSubNodes();
    }
  };

  componentDidMount() {
    this.props.fetchGraphs();
    this.props.fetchCollections();
    this.props.fetchEdges();
    this.props.fetchQueries();
    this.props.getServerData();
    document.addEventListener('keydown', _.throttle(this.handleKeyDown, 100));
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.handleKeyDown);
  }

  render() {
    return (
      <div className="main">
        <Sidebar />
        <TabContent tabKey="search">
          <Search />
        </TabContent>
        <TabContent tabKey="map">
          <Stage />
        </TabContent>
        <TabContent tabKey="favorites">
          <Favorites />
        </TabContent>
        <TabContent tabKey="reports">
          <Reports />
        </TabContent>
        <Loading iconSize="big"
                 isLoading={this.props.findLoading
                            || this.props.findReportLoading
                            || this.props.traversalLoading
                            || this.props.getSharedLoading
                            || this.props.saveUserMapLoading
                            || this.props.getUserMapLoading
                            || this.props.saveSharedLoading} />
        <Modal />
      </div>
    );
  }

}

function mapStateToProps(state) {
  return {
    findLoading: state.nodes.findLoading,
    findReportLoading: state.reports.loading,
    traversalLoading: state.nodes.traversalLoading,
    getSharedLoading: state.stage.getSharedLoading,
    saveUserMapLoading: state.stage.saveUserMapLoading,
    getUserMapLoading: state.stage.getUserMapLoading,
    saveSharedLoading: state.stage.saveSharedLoading
  };
}

export default connect(
  mapStateToProps,
  {
    fetchGraphs,
    fetchCollections,
    fetchEdges,
    fetchQueries,
    getServerData,
    clearCurrentNode,
    resetSubNodes,
    toggleHasId
  }
)(App);
