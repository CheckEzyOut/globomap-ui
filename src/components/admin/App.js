/*
Copyright 2017 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React, { Component } from 'react';
import './css/App.css';
import { TextInputGroup } from "./controls/Text";
import Properties from "./Properties"

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      node: {
        name: "",
        data: [{
          name: "",
          value: ""
        }]
      }
    };
  }

  setStateNode = (data) => {
    let newState = {};
    newState["node"] = data;
    this.setState(newState);
  }

  _changeCommon = (event) => {
    let key = event.target.attributes.name.nodeValue;
    let value = event.target.value;
    this.setState({
      [key]: value
    })
    // this.updateChartData({[key]: {$set: value}});
  }

  render() {
    return (
      <div className="main">
        <header className="main-header">
          <div className="header-group">
            <span className="logo">
              globo<span className="logo-map">map</span>
            </span>
          </div>
        </header>
        <div className={'admin-content'}>
          <div className="admin-content-top"></div>

          <div className="admin-content-middle">
            <TextInputGroup
              key="name"
              name="name"
              label="Nome"
              placeholder="Nome"
              onChange={this._changeCommon}
              defaultValue={this.state.node.name}
              />
            <Properties node={this.state.node} setStateNode={this.setStateNode} />
          </div>

          <div className="admin-content-base"></div>
        </div>
      </div>
    );
  }
}

export default App;
