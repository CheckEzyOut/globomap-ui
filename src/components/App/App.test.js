import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { App } from './App';

describe("App", function() {

  let componentDidMountStub;

  beforeEach(function() {
    componentDidMountStub = sinon.stub(App.prototype, 'componentDidMount').value('newValue');
  });

  afterEach(function () {
    componentDidMountStub.restore();
  });

  it('should render the App component', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

});
