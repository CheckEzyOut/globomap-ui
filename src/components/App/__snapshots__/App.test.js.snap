// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component 1`] = `
<div
  className="main"
>
  <Connect(Sidebar) />
  <Connect(TabContent)
    tabKey="search"
  >
    <Connect(Search) />
  </Connect(TabContent)>
  <Connect(TabContent)
    tabKey="map"
  >
    <Connect(Stage) />
  </Connect(TabContent)>
  <Connect(TabContent)
    tabKey="favorites"
  >
    <Connect(Favorites) />
  </Connect(TabContent)>
  <Loading
    iconSize="big"
  />
  <Connect(Modal) />
</div>
`;
